dashboard/app/controllers/projects_controller.rb

```diff
@@ -133,10 +133,59 @@ def public
     end
   end
 
+  # @param [Array {Hash}] Each hash is data for a row in the featured projects tables.
+  # The rows are sorted into two arrays: featured or unfeatured, based on
+  # on whether the project is currently featured or not.
+  def sort_projects(featured_project_table_rows)
+    @featured = []
+    @unfeatured = []
+    featured_project_table_rows.each do |row|
+      row[:featured] ? @featured << row : @unfeatured << row
+    end
+  end
+
+  # @param [Hash] key is a FeaturedProject, value is the associated project
+  # Iterate through the Featured Project - Project pairs are creates a new hash # for each containing combined information specific to what the featured
+  # projects tables are expecting.
+  def combine_projects_and_featured_projects_data(matched_projects_and_featured_projects)
+    @featured_project_table_rows = []
+    matched_projects_and_featured_projects.each do |featured_project, project|
+      project_details = JSON.parse(project[:id][:value])
+      featured_project_row = {
+        projectName: project_details['name'],
+        type: project_details['projectType'],
+        publishedAt: project[:id][:published_at],
+        thumbnailUrl: project_details['thumbnailUrl'],
+        featuredAt: featured_project.featured_at,
+        unfeaturedAt: featured_project.unfeatured_at,
+        featured: featured_project.featured?
+      }
```

Comment:
only the server can compute the encrypted channel id, so you will need to pass it to the client somewhere in this response. Here is where the encrypted channel id is computed in the other example: https://github.com/code-dot-org/code-dot-org/blob/418fb5b8d0cbfe4e63cdc2fa6ed2bea74b14b543/dashboard/lib/projects_list.rb#L139

to make this work in your case, you'll need the storage_id and id fields from PEGASUS_DB[:storage_apps] to be included in your SQL query/join.

At some point you'll need to construct the project share link from the encrypted channel id and the app type. It seems like you could do this either on the client or the server, but in the example I've been linking to it appears that we do this on the client: https://github.com/code-dot-org/code-dot-org/blob/2be1365483e4261faa6b53911b6415bcdb5b0e7d/apps/src/templates/projects/ProjectCard.jsx#L95